etcdctl/ctlv3/command/util.go

```diff
@@ -75,3 +79,43 @@ func commandCtx(cmd *cobra.Command) (context.Context, context.CancelFunc) {
 	}
 	return context.WithTimeout(context.Background(), timeOut)
 }
+
+// get the resident and virtual memory from <server:2379>/metrics
+func endpointMemoryMetrics(host string) (float64, float64) {
+	var residentMemory string
+	var virtualMemory string
+	if !strings.HasPrefix(host, `http://`) {
+		host = "http://" + host
+	}
+	url := host + "/metrics"
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Println(fmt.Sprintf("fetch error: %v", err))
+		return 0.0, 0.0
+	}
+	byts, readerr := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	if readerr != nil {
+		fmt.Println(fmt.Sprintf("fetch error: reading %s: %v", url, readerr))
+		return 0.0, 0.0
+	}
+
+	for _, line := range strings.Split(string(byts), "\n") {
+		if strings.HasPrefix(line, `process_resident_memory_bytes`) {
+			residentMemory = strings.TrimSpace(strings.TrimPrefix(line, `process_resident_memory_bytes`))
+		}
+		if strings.HasPrefix(line, `process_virtual_memory_bytes`) {
```

Comment:
virtual memory is not the total available memory on the host. i guess we need to modify etcd server to make it report the total available memory on host.

how about just keeping the percentage calculation as a todo, and simply reporting the memory usage and outputting pass always?

Reply:
@xiang90 sure, thanks. That sounds a great approach !! (I have a new go func created for darwin to find available memory using vm_stat (hw.physmem is not reliable at all) and one on Linux with syscall, not sure if we can use that for our todo or on server side but as you suggested let's take that separately for a concise and clear discussion on it's own). Thanks!